utf8proc.c

```diff
@@ -165,12 +168,9 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
     dst[0] = 0xC0 + (uc >> 6);
     dst[1] = 0x80 + (uc & 0x3F);
     return 2;
-  } else if (uc == 0xFFFF) {
-    dst[0] = 0xFF;
-    return 1;
-  } else if (uc == 0xFFFE) {
-    dst[0] = 0xFE;
-    return 1;
+  } else if (((utf8proc_uint32_t)uc)-0xd800 < 0x800) {
+    // Surrogates not valid code point, can't be encoded!
+    return 0;
```

Comment:
I meant for surrogates, which can appear from incorrect conversions of UTF-16 to UTF-8 (or Java-like "modified UTF-8"); I'm wondering what the tradeoffs here are for being permissive in the encoding step, so that we least provide the option of allowing such nonsense.

(We definitely want to detect them and throw an error *somewhere*, e.g. if `UTF8PROC_REJECTNA` is requested, but in that case it is optional.)